conda/lock.py

```diff
@@ -20,13 +20,31 @@
 import logging
 import os
 import time
+from glob import glob
+from os.path import abspath, isdir, join
 
+from conda.install import rm_rf
+from .compat import range
 from .exceptions import LockError
 
 LOCKFN = '.conda_lock'
 
+# Keep the string "LOCKERROR" in this string so that external
+# programs can look for it.
+LOCKSTR = """\
+LOCKERROR: It looks like conda is already doing something.
+The lock %s was found. Wait for it to finish before continuing.
+If you are sure that conda is not running, remove it and try again.
+You can also use: $ conda clean --lock
+"""
 
 stdoutlog = logging.getLogger('stdoutlog')
+log = logging.getLogger(__name__)
+
+def touch(fname, times=None):
+    # http://stackoverflow.com/a/1160227/2127762
+    with open(fname, 'a'):
+        os.utime(fname, times)
```

Comment:
Functions like this ultimately should be moved to a common utility library.  That'll happen in the near future.